### PR TITLE
Product Editor: Set description properly if description consists only of a single non-paragraph block

### LINKED ISSUE #45740: Product Editor: Description disappears if the only block is not a paragraph
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In the product description, if the first block is not a paragraph, whatever content has been added to the product description, will be lost.

### Expected behavior

The product description should be savable regardless of its content

### Actual behavior

The product description disappears when the first block is not a paragraph.

### Steps to reproduce

Go to WooCommece > product
add a new product.
in the product editor, open the deacription field
add a paragraph and save. You will see a oreview.
Reopen the description field and now set as a first block a column block
Then save the changes.
The product description will be gone.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): http://atelierdellaluce.com
Site address (URL): http://atelierdellaluce.com
WC Version: 8.7.0
REST API Version: ✔ 8.7.0
Action Scheduler Version: ✔ 3.7.2
Log Directory Writable: ✔
WP Version: 6.5-RC3
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: –
WP Cron: ✔
Language: de_DE
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.33
PHP Post Max Size: 256 MB
PHP Time Limit: 120
PHP Max Input Vars: 3000
cURL Version: 8.2.1
OpenSSL/3.1.1

SUHOSIN Installed: –
MySQL Version: 5.7.44-48-log
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.7.0
WC Database Prefix: bqf_
Total Database Size: 19.03MB
Database Data Size: 15.86MB
Database Index Size: 3.17MB
bqf_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_actionscheduler_actions: Data: 0.05MB + Index: 0.11MB + Engine InnoDB
bqf_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
bqf_jetpack_sync_queue: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_automations: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_automation_runs: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
bqf_mailpoet_automation_run_logs: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_mailpoet_automation_run_subjects: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_automation_triggers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_automation_versions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletters: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_statistics_bounces: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
bqf_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
bqf_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_subscribers: Data: 0.02MB + Index: 0.23MB + Engine InnoDB
bqf_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_subscriber_tag: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_mailpoet_tags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_user_agents: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_options: Data: 8.05MB + Index: 0.08MB + Engine InnoDB
bqf_postmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
bqf_posts: Data: 3.27MB + Index: 0.06MB + Engine InnoDB
bqf_sgs_log_events: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
bqf_sgs_log_visitors: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
bqf_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
bqf_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
bqf_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
bqf_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
bqf_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
bqf_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
bqf_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
bqf_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
bqf_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 14
mailpoet_page: 1
page: 7
post: 2
product: 8
revision: 164
shop_order_placehold: 2
wp_font_face: 33
wp_font_family: 10
wp_global_styles: 2
wp_navigation: 2
wp_template: 6
wp_template_part: 3

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (5) ###

Code Snippets: by Code Snippets Pro – 3.6.4
Speed Optimizer: by SiteGround – 7.4.6
Ajax add to cart for WooCommerce: by QuadLayers – 2.3.2
WooCommerce: by Automattic – 8.7.0
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.5.5

### Inactive Plugins (7) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.1
Gutenberg: by Gutenberg Team – 17.9.0
Jetpack: by Automattic – 13.2.1
MailPoet: by MailPoet – 4.46.0
Security Optimizer: by SiteGround – 1.4.12
WooCommerce Beta Tester: by WooCommerce – 2.3.0
WooPayments: by Automattic – 7.3.0

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 16 KB

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: ❌ Page does not contain the [woocommerce_checkout] shortcode or the woocommerce/checkout block.
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.0
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /home/customer/www/atelierdellaluce.com/public_html/wp-content/plugins/woocommerce/templates/block-notices/error.php
/home/customer/www/atelierdellaluce.com/public_html/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/home/customer/www/atelierdellaluce.com/public_html/wp-content/plugins/woocommerce/templates/block-notices/success.php

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-pre-publish-modal
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-03-20 14:23:10 +00:00
Options: ✔
Notes: 71
Onboarding: completed

### Action Scheduler ###

Complete: 95
Oldest: 2024-03-14 14:23:21 +0000
Newest: 2024-03-20 10:24:16 +0000

Pending: 1
Oldest: 2024-03-21 06:18:55 +0000
Newest: 2024-03-21 06:18:55 +0000

### Status report information ###

Generated at: 2024-03-20 10:32:21 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by mattsherman**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR fixes the blank description detection logic to account for the case where the description consists of a single non-paragraph block.

Closes #45740.

#### Before (description does not get set)

https://github.com/woocommerce/woocommerce/assets/2098816/e6e6f84f-e353-4777-a81f-a2625cf2c2c7

#### After (description gets set)

https://github.com/woocommerce/woocommerce/assets/2098816/c71601cf-9f5c-4f26-95d7-113f190f5f26

#### Automated testing

1. Run unit tests and verify they pass (you can just look at the results on this PR):

```
pnpm --filter=./packages/js/product-editor run test:js ./packages/js/product-editor/src/components/iframe-editor/utils/tests/are-blocks-empty.test.ts
```
#### Manual testing

With a WooCommerce env with the new product editor enabled (**WooCommerce** > **Settings** > **Advanced** > **Features** > **Experimental features**)...

1. Go to **Products** > **Add New**
2. Clear dev console in browser
3. Click on **Description** > **Full Editor** to bring up the modal block editor
4. Click **Done**
5. Verify the description is blank and can still be edited in the description field (not in the modal block editor)
6. Click on **Description** > **Full Editor** to bring up the modal block editor
7. Add a quote block
8. Click **Done**
9. Verify the description is set correctly (it should not be blank)

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/iframe-editor/utils/are-blocks-empty.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+import { BlockInstance } from '@wordpress/blocks';
+
+/**
+ * By default the blocks returned by the editor contains one paragraph
+ * block with empty content. This function checks to see if the blocks
+ * consists of a single paragraph block with empty content.
+ *
+ * @param blocks The block list
+ * @return true if the blocks consists of a single paragraph block with empty content; false otherwise
```
Comment (nathanss) [inline packages/js/product-editor/src/components/iframe-editor/utils/are-blocks-empty.ts]: This also returns true if there are no blocks, right? So the documentation is a bit confusing.

File: packages/js/product-editor/src/components/iframe-editor/utils/tests/are-blocks-empty.test.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+				innerBlocks: [],
+				isValid: true,
+				clientId: '1',
+			},
+		];
+
+		expect( areBlocksEmpty( blocks ) ).toBe( false );
+	} );
+
+	it( 'should return true if an empty block list is passed', () => {
```
Comment (nathanss) [inline packages/js/product-editor/src/components/iframe-editor/utils/tests/are-blocks-empty.test.ts:91]: Referring to the previous comment, this unit test seems to prove what I mentioned.

**Review by nathanss — APPROVED** [review]
Looking good and testing well! Just left a small comment. I'll pre-approve since it's a very small thing.

File: packages/js/product-editor/src/components/iframe-editor/utils/are-blocks-empty.ts (hunk 1 above)
Comment (mattsherman) [reply packages/js/product-editor/src/components/iframe-editor/utils/are-blocks-empty.ts]: Good catch. I'll update.

**Pull request merged**